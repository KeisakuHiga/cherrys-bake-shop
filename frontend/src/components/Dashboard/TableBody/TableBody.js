import React, { Component } from 'react';
import Moment from 'react-moment'
import style from './TableBody.module.css'

class TableBody extends Component {
  state = {  }
  render() {
    const { data, id } = this.props;
    return (
      <>
        <tr key={data._id} onClick={() => window.location =`/QuoteDetail/${data._id}`} className={style.tablerow}>
          <th key={data._id} scope="row" className="indexNumber">{ id + 1 }</th>
          <td key={data.user.userName.firstName} className="firstName" >{data.user.userName.firstName}</td>
          <td key={data.user.userName.lastName} className="lastName" >{data.user.userName.lastName}</td>
          <td key={data.user.contact.phoneNumber} className="phoneNumber" >{data.user.contact.phoneNumber}</td>
          <td key={data.pickUpDateAndTime}><Moment local format="MMM DD, YYYY LT">{data.pickUpDateAndTime}</Moment></td>
          <td key={data.createdAt}><Moment local format="MMM DD, YYYY LT">{data.createdAt}</Moment></td>
        </tr>
      </>
    );
  }
}

export default TableBody;