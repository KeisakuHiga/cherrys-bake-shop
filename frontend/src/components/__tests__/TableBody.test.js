import React from "react";
import Enzyme, { shallow, render, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { createSerializer } from "enzyme-to-json";
import sinon, { spy } from "sinon";

// Set the default serializer for Jest to be the from enzyme-to-json
// This produces an easier to read (for humans) serialized format.
expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });

// import chai, { expect } from "chai";
// import sinonChai from "sinon-chai";
// chai.use(sinonChai)

import TableBody from "../Dashboard/TableBody/TableBody"

describe("<TableBody />", () => {
  
  let passedData = {
    _id: 'dlakjkakl;akl15678dja',
    user: {
      userName:{ firstName: '', lastName: '' },
      contact: { phoneNumber: ''},
      pickUpDateAndTime: '',
      createdAt: ''
    }
  }

  it('renders component when there is no data', () => {
    const wrapper = shallow(<TableBody data={passedData} />)
    expect(wrapper).toMatchSnapshot()
  })

  passedData = {
    _id: '194jdalk1l2lda1jk1',
    user: {
      userName:{ firstName: 'John', lastName: 'Smith' },
      contact: { phoneNumber: '0456753913'},
      pickUpDateAndTime: '2019-08-08T12:30',
      createdAt: '2019-08-01T10:50'
    }
  }
  it('renders component when there is a quotes data', () => {
    const wrapper = shallow(<TableBody data={passedData} />)
    expect(wrapper).toMatchSnapshot()
  })

  it("correctly formats the first name", () => {
    const wrapper = shallow(<TableBody data={passedData} />)
    const text = wrapper.find(".firstName").text();
    expect(text).toEqual('John');
  });

  it("correctly formats the last name", () => {
    const wrapper = shallow(<TableBody data={passedData} />)
    const text = wrapper.find(".lastName").text();
    expect(text).toEqual('Smith');
  });

  it("correctly formats the phone number", () => {
    const wrapper = shallow(<TableBody data={passedData} />)
    const text = wrapper.find(".phoneNumber").text();
    expect(text).toEqual('0456753913');
  });

  // it("correctly formats the pick up time and date", () => {
  //   const wrapper = render(<Moment />)
  //   const text = wrapper.find(".pickUpDateAndTime").text();
  //   expect(text).toEqual('Aug 08, 2019 12:30 PM');
  // });
});

