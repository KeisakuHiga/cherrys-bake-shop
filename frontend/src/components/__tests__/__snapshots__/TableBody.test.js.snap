// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableBody /> renders component when there is a quotes data 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`<TableBody /> renders component when there is no data 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
